python/sglang/srt/managers/schedule_policy.py

```diff
@@ -51,18 +60,50 @@ def calc_priority(self, waiting_queue: List[Req]):
 
         # Compute matched prefix length
         prefix_computed = False
+        # rid to deprioritize in the current run.
+        temporary_deprioritized = {}
         if policy == "lpm" or policy == "dfs-weight":
+            # It is used to find the matching prefix for in-batch prefix caching.
+            temp_radix = RadixCache(None, None, False)
             for r in waiting_queue:
+                prefix_ids = r.adjust_max_prefix_ids()
                 # NOTE: the prefix_indices must always be aligned with last_node
                 r.prefix_indices, r.last_node = self.tree_cache.match_prefix(
-                    rid=r.rid, key=r.adjust_max_prefix_ids()
+                    rid=r.rid, key=prefix_ids
                 )
 
+                # NOTE(sang): This logic is for In-batch prefix caching;
+                # If there are more than 1 request that have small matching prefix from
+                # existing cache, but all those requests share the same prefix, we prefer
+                # to schedule only one of them so that we can increase the cache hit rate.
+                # We prefer to set IN_BATCH_PREFIX_CACHING_THRESHOLD > 0 because too small
```

Comment:
> IN_BATCH_PREFIX_CACHING_THRESHOLD > 0` 

why >0? should this be > 32?

> It is kind of common when the engine is long running (e.g., imagine "the"). 

What does imagine "the" mean?

Reply:
if the threshold is 0, this optimization is not applied to prefix like "the", which is common. 

Regarding the comment, I just meant == 0 is not ideal because it misses cases like "the" prefix

Reply:
32 is also an arbitrary value actually. didn't do much tuning here

Reply:
Makes sense